keps/sig-scheduling/5471-enable-sla-based-scheduling/README.md

```diff
@@ -0,0 +1,1430 @@
+    effect: NoSchedule
+  containers:
+  - name: ml-workload
+    resources:
+      claims:
+      - name: gpu-claim
+```
+
+#### Story 5 — DRA device-level error budget management
+
+As a platform engineer managing GPU clusters with varying reliability states, I want to allocate devices based on their remaining error budget using numeric tolerations. So that critical workloads only get devices with sufficient reliability headroom while allowing degraded devices to serve less sensitive workloads.
+
+This will get the critical inference fresh devices (>24h error budget), batch training can use aging devices (1-24h), and severely degraded devices (<1h) are excluded from allocation entirely, enabling graceful device lifecycle management.
+
+**Example Configuration:**
+
+```yaml
+# Driver taints devices with low error budget
+kind: ResourceSlice
+spec:
+  driver: device.example.com
+  devices:
+  - name: gpu-node-01-device-0
+    attributes:
+      memory: "32Gi"
+      compute-capability: "8.6"
+    # Driver applies taint when error budget drops below 10 hours
+    taints:
+    - key: device.example.com/error-budget-in-hours
+      value: "8"  # 8 hours remaining
+      effect: NoSchedule
+---
+# Critical inference workload requires high-reliability devices
+kind: ResourceClaim
+metadata:
+  name: inference-gpu-claim
+spec:
+  requests:
+  - name: high-reliability-gpu
+    deviceClassName: device.example.com
+    tolerations:
+    # Only accept devices with >24 hours error budget
+    - key: device.example.com/error-budget-in-hours
+      operator: Gt
+      value: "24"
+      effect: NoSchedule
+---
+# Batch training workload tolerates degraded devices
+kind: ResourceClaim
+metadata:
+  name: training-gpu-claim
+spec:
+  requests:
+  - name: batch-gpu
+    deviceClassName: device.example.com
+    tolerations:
+    # Accept devices with >1 hour error budget
+    - key: device.example.com/error-budget-in-hours
+      operator: Gt
+      value: "1"
+      effect: NoSchedule
+```
+
+#### Story 6 — Kubernetes version compatibility for critical workloads
+
+As a cluster operator managing a mixed-version Kubernetes cluster during rolling upgrades, I want to ensure critical workloads only run on nodes with Kubernetes version >= 1.20.0 due to specific API features they require, while allowing development workloads to tolerate older versions.
+
+Semantic version tolerations provide precise version-based placement without hardcoding specific versions in NodeAffinity rules.
+
+**Example Configuration:**
+
+```yaml
+# Older node running Kubernetes 1.19.5
+apiVersion: v1
+kind: Node
+metadata:
+  name: old-node-1
+spec:
+  taints:
+  - key: node.kubernetes.io/version
+    value: "1.19.5"
+    effect: NoSchedule
+---
+# Production workload requires Kubernetes >= 1.20.0
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: production-api
+spec:
+  template:
+    spec:
+      tolerations:
+      - key: node.kubernetes.io/version
+        operator: SemverGt
```

Comment:
SemverGe for this and below in this example?

Reply:
Deffered to the future KEP for semver/CEL